### PR TITLE
Add `TensorMesh.cell_bounds` property

### PR DESCRIPTION
**Pull request by santisoler**
Add a new `cell_bounds` property to `TensorMesh` that returns the bounds of each cell in the mesh. This implements a faster way to obtain the bounds of each cell in the mesh than trying to access them through `TensorCell.bounds`.

### CODE REVIEW

**Comment by jcapriot** [comment]
don't worry too much about updating the CI build stuff at the moment, I've almost got it put back together.

File: discretize/tensor_mesh.py [hunk 1]
```diff
@@ -593,6 +593,36 @@ def face_boundary_indices(self):
...
+        else:
+            x1 = centers[:, 0] - widths[:, 0] / 2
+            x2 = centers[:, 0] + widths[:, 0] / 2
+            y1 = centers[:, 1] - widths[:, 1] / 2
+            y2 = centers[:, 1] + widths[:, 1] / 2
+            z1 = centers[:, 2] - widths[:, 2] / 2
+            z2 = centers[:, 2] + widths[:, 2] / 2
+            bounds = (x1, x2, y1, y2, z1, z2)
+
+        cell_bounds = np.hstack(tuple(v[:, np.newaxis] for v in bounds))
```
Comment (jcapriot) [inline discretize/tensor_mesh.py:623]: I would suggest this as a cleaner way to do it...

```suggestion
        nodes = self.nodes.reshape((*self.shape_nodes, -1), order="F")
    
        min_nodes = nodes[(slice(-1), )*self.dim]
        min_nodes = min_nodes.reshape((self.n_cells, -1), order="F")
        max_nodes = nodes[(slice(1, None), )*self.dim]
        max_nodes = max_nodes.reshape((self.n_cells, -1), order="F")
    
        cell_bounds = np.stack((min_nodes, max_nodes), axis=-1)
        cell_bounds = cell_bounds.reshape((self.n_cells, -1))
```

**Review by jcapriot — CHANGES_REQUESTED** [review]

File: discretize/tensor_mesh.py (hunk 1 above)
Comment (santisoler) [reply discretize/tensor_mesh.py:623]: Looks good!

**Comment by santisoler** [comment]
@jcapriot I think this is ready. Feel free to merge if you think so.

File: discretize/tensor_mesh.py (hunk 1 above)
Comment (jcapriot) [reply discretize/tensor_mesh.py]: Awesome, just trying to reduce any potential accumulated floating point errors.

**Pull request merged**